schema/Metrics.v1.0.0.schema.json

```diff
@@ -0,0 +1,354 @@
+      "minLength": 1,
+      "maxLength": 200
+    },
+    "description": {
+      "type": "string",
+      "description": "A description of the metric.",
+      "minLength": 1,
+      "maxLength": 1000
+    },
+    "tags": {
+      "type": "array",
+      "description": "The tags of the metric.",
+      "items": {
+        "type": "string",
+        "description": "A tag attached to the metric.",
+        "minLength": 1,
+        "maxLength": 30,
+        "pattern": "^[a-zA-Z0-9 _:()&-]+$"
+      }
+    },
+    "desiredDirection": {
+      "description": "Whether an increase or decrease to the metric value is desired.",
+      "type": "string",
+      "enum": ["Increase", "Decrease", "Neutral"],
+      "x-ms-enum": {
+        "name": "DesiredDirection",
+        "modelAsString": true,
+        "values": [
+          {
+            "name": "Increase",
+            "value": "Increase",
+            "description": "An increase to the metric value is desired. E.g., Success rate, Total revenue."
+          },
+          {
+            "name": "Decrease",
+            "value": "Decrease",
+            "description": "A decrease to the metric value is desired. E.g., Error rate, Latency."
+          },
+          {
+            "name": "Neutral",
+            "value": "Neutral",
+            "description": "Neither an increase nor a decrease to the metric value is desired, or the desired direction depends on other factors. E.g., Number of video play attempts, Number of user feedbacks"
+          }
+        ]
+      }
+    },
+    "definition": {
+      "$ref": "#/definitions/MetricDefinition"
+    }
+  },
+  "definitions": {
+    "EventCountDefinition": {
+      "type": "object",
+      "description": "The definition of an EventCount metric. This metric kind counts the observations of an event. Experiment analysis accounts for unequal traffic allocation.",
+      "properties": {
+        "event": {
+          "$ref": "#/definitions/ObservedEvent",
+          "description": "Event to observe."
+        }
+      },
+      "required": ["event"],
+      "x-ms-discriminator-value": "EventCount"
+    },
+    "UserCountDefinition": {
+      "type": "object",
+      "description": "The definition of a UserCount metric. This metric kind counts the users who encounter an event. Experiment analysis accounts for unequal traffic allocation.",
+      "properties": {
+        "event": {
+          "$ref": "#/definitions/ObservedEvent",
+          "description": "Event to observe."
+        }
+      },
+      "required": ["event"],
+      "x-ms-discriminator-value": "UserCount"
+    },
+    "EventRateDefinition": {
+      "type": "object",
+      "required": ["kind", "event", "condition"],
+      "properties": {
+        "kind": {
+          "type": "string",
+          "enum": ["EventRate"],
+          "description": "The kind of metric."
+        },
+        "event": {
+          "allOf": [
```

Comment:
The `allOf` pattern is reused throughout this JSON schema, but is it needed when there's only 1 ref class?

Reply:
Its not needed. Removed all "allOf"